### PR TITLE
IPv6 dual stack config

### PR DESCRIPTION
**Pull request by buuhsmead**
## Description
Moved into default vars the network settings for ipv4 and ipv6
Add / remove ipv6 entries at transip dns provider

## Checklist/ToDo's

- [X ] Added documentation?  docs/ipv6.md

- [ ] Added release note entry? (  docs/release-notes.md )

- [ X] Tested? simple test, works on my machine

### CODE REVIEW

File: ansible/roles/openshift-4-cluster/templates/install-config.yaml.j2 [hunk 1]
```diff
@@ -11,28 +11,28 @@ controlPlane:
...
+    hostPrefix: {{ networking_clusternetwork_hostprefix_ipv6 }}
 {% endif %}
   machineNetwork:
 {% if 'IPv4' in ip_families  %}
-  - cidr: {{ vn_subnet }}/24
+  - cidr: {{ networking_machinenetwork_cidr_ipv4 }}
 {% endif %}
 {% if 'IPv6' in ip_families  %}
-  - cidr: {{ vn_subnet_ipv6 }}::/80
+  - cidr: {{ networking_machinenetwork_cidr_ipv6 }}
```
Comment (rbo) [inline ansible/roles/openshift-4-cluster/templates/install-config.yaml.j2:28]: Why not using  `vn_subnet_ipv6`? Is there any reason to use a different `networking_machinenetwork_cidr_ipv6`?

File: ansible/roles/openshift-4-cluster/templates/install-config.yaml.j2 [hunk 2]
```diff
@@ -11,28 +11,28 @@ controlPlane:
...
 {% if 'IPv6' in ip_families  %}
-  - cidr: fd00:10:128::/56
-    hostPrefix: 64
+  - cidr: {{ networking_clusternetwork_cidr_ipv6 }}
+    hostPrefix: {{ networking_clusternetwork_hostprefix_ipv6 }}
 {% endif %}
   machineNetwork:
 {% if 'IPv4' in ip_families  %}
-  - cidr: {{ vn_subnet }}/24
+  - cidr: {{ networking_machinenetwork_cidr_ipv4 }}
```
Comment (rbo) [inline ansible/roles/openshift-4-cluster/templates/install-config.yaml.j2:25]: Why not using  `vn_subnet`? Is there any reason to use a different `networking_machinenetwork_cidr_ipv4`?

File: ansible/roles/openshift-4-cluster/templates/install-config.yaml.j2 [hunk 3]
```diff
@@ -11,28 +11,28 @@ controlPlane:
...
-  - cidr: 10.128.0.0/14
-    hostSubnetLength: 9
+  - cidr: {{ networking_clusternetwork_cidr_ipv4 }}
+    hostPrefix: {{ networking_clusternetwork_hostprefix_ipv4 }}
 {% endif %}
 {% if 'IPv6' in ip_families  %}
-  - cidr: fd00:10:128::/56
-    hostPrefix: 64
+  - cidr: {{ networking_clusternetwork_cidr_ipv6 }}
+    hostPrefix: {{ networking_clusternetwork_hostprefix_ipv6 }}
```
Comment (rbo) [inline ansible/roles/openshift-4-cluster/templates/install-config.yaml.j2:21]: Is the `hostPrefix` for IPv6 really configurable? I guess /64 is the only supported option, right?

File: ansible/roles/openshift-4-cluster/templates/install-config.yaml.j2 (hunk 1 above)
Comment (buuhsmead) [reply ansible/roles/openshift-4-cluster/templates/install-config.yaml.j2:28]: When using 'vn_subnet_ipv6' the nodes are directly accessible from the Internet. Something I did not like. So used a different subnet, not reachable from the Internet.

Comment (buuhsmead) [reply ansible/roles/openshift-4-cluster/templates/install-config.yaml.j2:28]: That would also be a solution. Which I have not yet found. And by using the current fd02:: it is even save when making mistakes. 
If you don't like this. I can take more time and look for more Ansible rules for using vn_net_ipv6::/80.

File: ansible/roles/openshift-4-cluster/templates/install-config.yaml.j2 (hunk 2 above)
Comment (buuhsmead) [reply ansible/roles/openshift-4-cluster/templates/install-config.yaml.j2:25]: Had another look on this item. Actually '{{ networking_machinenetwork_cidr_ipv4 }}' is the same as '{{ vn_subnet }}/24'. It is only moved to the defaults main.yaml settings. The var named can be changed into 'vn_subnet_cidr_ipv4' to state that

**Comment by rbo** [comment]
With commit 9df66400abbbb715aa998b3866e713cf876093e5 in devel tree I fixed the DigitalOcean DNS provider (IPv4 & IPv6). I switch to ` community.digitalocean.digital_ocean_domain_record` it's indeed better. Thanks for the hint.

**Pull request closed without merging**